submissions/tests/test_team_api.py

```diff
@@ -0,0 +1,401 @@
+        team_submission_uuid = result.pop('team_submission_uuid')
+        submission_uuids = result.pop('submission_uuids')
+
+        self.assertDictEqual(
+            {
+                'submitted_at': now(),
+                'created_at': now(),
+                'attempt_number': 1,
+                'course_id': COURSE_ID,
+                'item_id': ITEM_1_ID,
+                'team_id': TEAM_1_ID,
+                'submitted_by': self.user_1.id
+            },
+            result
+        )
+        # Make sure the model was created
+        TeamSubmission.objects.get(uuid=team_submission_uuid)
+
+        # Make sure the submisisons have been created
+        self.assertEqual(len(submission_uuids), len(self.user_ids))
+        remaining_users = {str(user_id) for user_id in self.user_ids}
+        for submission_uuid in submission_uuids:
+            submission = Submission.objects.select_related('student_item').get(uuid=submission_uuid)
+            self.assertIn(submission.student_item.student_id, remaining_users)
+            self.assertEqual(COURSE_ID, submission.student_item.course_id)
+            self.assertEqual(ITEM_1_ID, submission.student_item.item_id)
+            self.assertEqual(result['attempt_number'], submission.attempt_number)
+            self.assertEqual(ANSWER, submission.answer)
+            remaining_users.remove(submission.student_item.student_id)
+
+    def test_create_submission_for_team_existing_active_team_submission(self):
+        """
+        Test for calling create_submission_for_team with an existing active team submission
+        for the same team course and item.
+        """
+        self.make_team_submission(status=ACTIVE)
+        with self.assertRaises(DuplicateTeamSubmissionsError):
+            self._call_create_submission_for_team_with_default_args()
+
+    def test_create_submission_for_team_existing_deleted_team_submission(self):
+        """
+        Test for calling create_submission_for_team with an existing deleted team submission
+        for the same team course and item.
+        """
+        team_submission_1 = self.make_team_submission(status=DELETED)
+        team_submission_2 = self._call_create_submission_for_team_with_default_args()
+        self.assertEqual(team_submission_1.attempt_number, 1)
+        self.assertEqual(team_submission_2['attempt_number'], 1)
+
+    def test_create_submission_for_team_invalid_attempt(self):
+        """
+        Test for calling create_submission_for_team with an invalid attempt_number
+        """
+        with self.assertRaises(TeamSubmissionRequestError):
+            team_api.create_submission_for_team(
+                COURSE_ID,
+                ITEM_1_ID,
+                TEAM_1_ID,
+                self.user_1.id,
+                self.user_ids,
+                ANSWER,
+                attempt_number=-1
+            )
+
+    def test_create_submission_for_team_existing_individual_submission(self):
+        """
```

Comment:
Can you explain what the behavior here is?

Reply:
I don't think this will ever happen, but in case we ever get into a weird data state I wanted this behavior explicitly demonstrated in a test.

For an individual submission, if we call create_submisison when there is an existing submission, we will increment the attempt_number of the most recent submission. For teams, though, we pass the `attempt_number` from `create_team_submission` to `create_submission` so that all of the numbers match. I don't know that it's .... important? It's just kinda weird so I wanted to be explicit.

Reply:
I'll expand this comment